### PR TITLE
(imp) SD-1810

### PR DESCRIPTION
**Pull request by sivakuna-aap**

### CODE REVIEW

**Comment by amagdas** [comment]
Needs a rebase...

**Comment by sivakuna-aap** [comment]
Done...but on travis it fails because of failures in installing the packages...restarting the build should fix the build. @amagdas can you please restart the build on travis?

**Comment by amagdas** [comment]
Done

**Comment by akintolga** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:

**Comment by ioanpocol** [comment]
:+1:

**Comment by sivakuna-aap** [comment]
@superdesk/developers @akintolga @sivakuna-aap @marwoodandrew @mdhaman 
If everyone is happy then I will merge this PR.

File: server/apps/users/services.py [hunk 1]
```diff
@@ -90,6 +90,51 @@ def get_invisible_stages(user):
...
+        Checks if the requested 'PATCH' or 'DELETE' operation is Invalid.
+        Operation is invalid if one of the below is True:
+            1. Check if the user is updating his/her own status.
+            2. Check if the user is changing the status of other logged-in users.
+            3. A user without 'User Management' privilege is changing role/user_type/privileges
+
+        :return: error message if invalid.
+        """
+
+        if 'user' in flask.g:
```
Comment (mugurrus) [inline server/apps/users/services.py]: Tight coupling with Flask is not a good idea. We should have our own user/session API and build an adapter for Flask.

File: server/superdesk/errors.py [hunk 2]
```diff
@@ -122,7 +122,14 @@ class UserInactiveError(SuperdeskApiError):
     """User is inactive, access restricted"""
     status_code = 403
     payload = {'is_active': False}
-    message = 'Account suspended, access restricted.'
+    message = 'Account is inactive, access restricted.'
+
+
+class UserDisabledError(SuperdeskApiError):
```
Comment (mugurrus) [inline server/superdesk/errors.py]: Errors specific to endpoints should be defined in their corresponding package. Now if you remove ingest for example the errors are still there.

File: server/apps/users/services.py (hunk 1 above)
Comment (sivakuna-aap) [inline server/apps/users/services.py]: Agreed, but `auth.py` sticks user to `flask.g` and that's the reason we had to use `flask.g`

**Comment by mugurrus** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:

**Pull request merged**